zerver/views/messages.py

```diff
@@ -622,17 +622,21 @@ def get_old_messages_backend(request, user_profile,
         after_query = query.where(inner_msg_id_col >= anchor) \
                            .order_by(inner_msg_id_col.asc()).limit(num_after)
 
-    if num_before == 0 and num_after == 0:
-        # This can happen when a narrow is specified.
-        after_query = query.where(inner_msg_id_col == anchor)
+    if anchor == 10000000000000000:
```

Comment:
It might be worth to have a constant here.  ABSURDLY_LARGE_MSG_ID = 1000000000000000000